### ISSUE TITLE
Attributes appearing on withdraws

### ISSUE THREAD

**Issue by spakka** [Exa-Networks/exabgp#272]
Hi,

After running for a while with lots of incoming updates (from a full-table feed), we start to see withdraw update messages which have attributes on them, despite the underlying message not having them (normally a withdraw just contains a list of prefixes unless there is also an announce in the same message)

We see the following example output:
```
{ "exabgp": "3.4.6", "time": 1432043305.52, "host" : "rrc-test.ripe.net", "pid" : "6388", "ppid" : "1", "type": "update", "header": "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF001F02", "body": "0008156D46A0145113700000", "neighbor": { "ip": "77.72.229.74", "address": { "local": "193.0.19.22", "peer": "77.72.229.74"}, "asn": { "local": "65000", "peer": "12654"}, "message": { "update": { "attribute": { "origin": "igp", "as-path": [ 12654, 9002, 57023, 36947, 36947 ], "confederation-path": [], "atomic-aggregate": false }, "withdraw": { "ipv4 unicast": { "109.70.160.0/21": {  }, "81.19.112.0/20": {  } } } } }} }
```

You can see there is an as-path, origin, etc on a withdraw. Yet, if I take the same 'header' and 'body', and put it back into exabgp --decode:
```
/opt/exabgp/bin/python2.6 /opt/exabgp/lib/python2.6/site-packages/exabgp/application/bgp.py --decode FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF001F020008156D46A0145113700000 /opt/exabgp/etc/exabgp/exabgp_instance_6.conf`
<snip>
Tue, 19 May 2015 14:40:43 | INFO     | 11368  | parser        | the message is an update
Tue, 19 May 2015 14:40:43 | INFO     | 11368  | parser        | no announced NLRI
Tue, 19 May 2015 14:40:43 | INFO     | 11368  | parser        | parsing 1/1 nlri payload 156D 46A0 1451 1370
Tue, 19 May 2015 14:40:43 | INFO     | 11368  | parser        | parsed withdraw nlri 109.70.160.0/21 next-hop none payload 156D 46A0
Tue, 19 May 2015 14:40:43 | INFO     | 11368  | parser        | parsing 1/1 nlri payload 1451 1370
Tue, 19 May 2015 14:40:43 | INFO     | 11368  | parser        | parsed withdraw nlri 81.19.112.0/20 next-hop none payload 1451 1370
Tue, 19 May 2015 14:40:43 | INFO     | 11368  | parser        |
Tue, 19 May 2015 14:40:43 | INFO     | 11368  | parser        | decoded update withdrawn 109.70.160.0/21 next-hop none
Tue, 19 May 2015 14:40:43 | INFO     | 11368  | parser        | decoded update withdrawn 81.19.112.0/20 next-hop none
Tue, 19 May 2015 14:32:45 | INFO     | 10871  | parser        | update json { "exabgp": "3.4.0", "time": 1432045965, "host" : "rrc-test.ripe.net", "pid" : "10871", "ppid" : "6144", "type": "update", "neighbor": { "ip": "77.72.229.74", "address": { "local": "193.0.19.22", "peer": "77.72.229.74"}, "asn": { "local": "65000", "peer": "12654"}, "message": { "update": { "withdraw": { "ipv4 unicast": { "109.70.160.0/21": {  }, "81.19.112.0/20": {  } } } } }} }
```

(This is for version 3.4.6 btw)

The same message on the wire is decoded by TShark as follows:
```
Border Gateway Protocol - UPDATE Message
    Marker: ffffffffffffffffffffffffffffffff
    Length: 31
    Type: UPDATE Message (2)
    Unfeasible routes length: 8 bytes
    Withdrawn routes:
        109.70.160.0/21
            Withdrawn route prefix length: 21
            Withdrawn prefix: 109.70.160.0 (109.70.160.0)
        81.19.112.0/20
            Withdrawn route prefix length: 20
            Withdrawn prefix: 81.19.112.0 (81.19.112.0)
    Total path attribute length: 0 bytes
```

I'd guess it is an attribute caching issue - we haven't managed to capture this happening where we have both an announce and a subsequent withdraw to compare the attributes. Also, it doesn't happen all the time!

If you need any extra info, please let me know

Thanks,
Colin

**Comment by thomas-mangin** [comment]
Hi Colin, I will investigate and fix in 3.5 - not sure I should in 3.4 ...

**Comment by thomas-mangin** [comment]
Colin, I had to think twice but I think I have a strong educated guess about what it could be.

Could you apply this patch and tell me if the issue still persist.
https://github.com/Exa-Networks/exabgp/commit/0f5449aff0b78a194b0d1734e1871d5ba69ad4bf
